app/api/routes/image_queries.py

```diff
@@ -108,26 +148,54 @@ async def post_image_query(
         },
     )
 
+    img_numpy = np.asarray(image)  # [H, W, C=3], dtype: uint8, RGB format
+
+    edge_inference_manager = app_state.edge_inference_manager
+    motion_detection_manager = app_state.motion_detection_manager
+    require_human_review = human_review == "ALWAYS"
+
     # TODO: instead of just forwarding want_async calls to the cloud, facilitate partial
     #       processing of the async request on the edge before escalating to the cloud.
     _want_async = want_async is not None and want_async.lower() == "true"
     if _want_async:
-        return safe_call_api(
-            gl.submit_image_query,
-            detector=detector_id,
-            image=image,
-            wait=0,
+        edge_only = os.environ.get("EDGE_ONLY", "") == "ENABLED"
+        if not edge_only:
+            # Escalate to cloud api as normal
+            return safe_call_api(
+                gl.submit_image_query,
+                detector=detector_id,
+                image=image,
+                wait=0,
+                patience_time=patience_time,
+                confidence_threshold=confidence_threshold,
+                human_review=human_review,
+                want_async=True,
+            )
```

Comment:
This preserves the same functionality as before when EDGE_ONLY is not set to ENABLED.